### PR TITLE
ci: add IAM roles for BYOID

### PR DESCRIPTION
**Pull request by sdowell**
This adds the IAM roles for RootSync and a RepoSync reconciler to use the k8sserviceaccount auth type.

### CODE REVIEW

File: e2e/testinfra/terraform/modules/service_account/service_account.tf [hunk 1]
```diff
@@ -40,3 +41,16 @@ resource "google_project_iam_member" "gcp_sa_role" {
...
 }
+
+// Create IAM bindings directly to the KSA(s) for BYOID
+resource "google_project_iam_member" "k8s_sa_role_rootsync" {
+  role    = var.role
+  member  = "serviceAccount:${data.google_project.project.project_id}.svc.id.goog[config-management-system/root-reconciler]"
+  project = data.google_project.project.id
+}
+
+resource "google_project_iam_member" "k8s_sa_role_reposync" {
```
Comment (nan-yu) [inline e2e/testinfra/terraform/modules/service_account/service_account.tf:52]: This isn't used right now. We can add it when we add test coverage for RepoSync with k8sserviceaccount.

File: e2e/testinfra/terraform/modules/service_account/service_account.tf [hunk 2]
```diff
@@ -40,3 +41,16 @@ resource "google_project_iam_member" "gcp_sa_role" {
   member  = "serviceAccount:${google_service_account.gcp_sa.email}"
   project = data.google_project.project.id
 }
+
+// Create IAM bindings directly to the KSA(s) for BYOID
+resource "google_project_iam_member" "k8s_sa_role_rootsync" {
```
Comment (nan-yu) [inline e2e/testinfra/terraform/modules/service_account/service_account.tf:46]: Technically, this is not associated with the GSA. I would rather it to be outside of the service account module.

It also needs an additional workload identity pool of the fleet host project (cs-dev-hub).

**Review by nan-yu — COMMENTED** [review]
I created #1227 for the missing IAM.

**Comment by sdowell** [comment]
Duplicate of https://github.com/GoogleContainerTools/kpt-config-sync/pull/1127

**Pull request closed without merging**